topology/probes/istio/gateway.go

```diff
@@ -44,3 +45,19 @@ func (h *gatewayHandler) Dump(obj interface{}) string {
 func newGatewayProbe(client interface{}, g *graph.Graph) k8s.Subprobe {
 	return k8s.NewResourceCache(client.(*kiali.IstioClient).GetIstioNetworkingApi(), &kiali.Gateway{}, "gateways", g, &gatewayHandler{})
 }
+
+func gatewayVirtualServiceAreLinked(a, b interface{}) bool {
+	gateway := a.(*kiali.Gateway)
+	vs := b.(*kiali.VirtualService)
+	vsGateways := vs.Spec["gateways"].([]interface{})
```

Comment:
You can't not "gateways" exists - please verify it does before using it - see how I created a virtualservice which is missing the gateways field:

```
[vagrant@dev istio]$ git diff
diff --git a/tests/istio/gateway-virtualservice.yaml b/tests/istio/gateway-virtualservice.yaml
index 874827ad..63213d04 100644
--- a/tests/istio/gateway-virtualservice.yaml
+++ b/tests/istio/gateway-virtualservice.yaml
@@ -20,11 +20,10 @@ metadata:
 spec:
   hosts:
   - reviews.prod.svc.cluster.local
-  gateways:
-  - skydive-test-gateway-virtualservice
+  #gateways:
+  #- skydive-test-gateway-virtualservice
   http:
   - route:
     - destination:
         host: reviews.prod.svc.cluster.local
         subset: v1
-
[vagrant@dev istio]$ kubectl describe gateway/skydive-test-gateway-virtualservice
Name:         skydive-test-gateway-virtualservice
Namespace:    default
Labels:       <none>
Annotations:  kubectl.kubernetes.io/last-applied-configuration={"apiVersion":"networking.istio.io/v1alpha3","kind":"Gateway","metadata":{"annotations":{},"name":"skydive-test-gateway-virtualservice","namespace":"de...
API Version:  networking.istio.io/v1alpha3
Kind:         Gateway
Metadata:
  Cluster Name:
  Creation Timestamp:  2019-02-24T06:59:00Z
  Generation:          1
  Resource Version:    247461
  Self Link:           /apis/networking.istio.io/v1alpha3/namespaces/default/gateways/skydive-test-gateway-virtualservice
  UID:                 a9af92dc-3801-11e9-a066-5254001c08b7
Spec:
  Selector:
    Istio:  ingressgateway
  Servers:
    Hosts:
      *
    Port:
      Name:      http
      Number:    80
      Protocol:  HTTP
Events:          <none>
[vagrant@dev istio]$
```